tensorflow/compiler/tf2tensorrt/convert/ops/fill_ops.cc

```diff
@@ -132,58 +132,72 @@ class ConvertRange : public ConvertFillBase<ConvertRange> {
-        return errors::Unimplemented(convert_range_expected_msg(node_def),
-                                     "tensors");
+    if (!(all_weights_ || all_integers_)) {
+      // As of 06/03/2022, when at least one of the (start, limit, delta)
+      // is passed as a tensor, they must all be of type kINT32
+      return errors::Unimplemented(convert_range_expected_msg(node_def));
+    }
+
+    if (inputs.at(2).is_weights()) {
+      if ((delta_ = param[2]) == 0) {
+        return errors::InvalidArgument("The delta parameter of ", node_def.op(),
+                                       " operation cannot be equal to 0");
       }
 
-      for (int i = 0; i < 3; i++) {
-        const auto& dims = inputs.at(i).GetTrtDims();
-        if (dims.nbDims != 1 || dims.d[0] != 1) {
-          return errors::InvalidArgument("Dimension for '", InputSpec()[i].name,
-                                         "' of ", node_def.op(), " operator ",
-                                         "should be equal to 1");
-        }
+      if (!all_weights_ && delta_ < 0) {
+        return errors::InvalidArgument(
+            "The delta parameter of Range operation "
+            "cannot be negative, when one of (start, limit) is passed as "
+            "a tensor, but got ",
+            delta_);
       }
-      return Status::OK();
     }
 
-    float param[3];
     for (int i = 0; i < 3; i++) {
       const auto& input = inputs.at(i);
-      switch (input.TrtDType()) {
-        case nvinfer1::DataType::kFLOAT:
-          param[i] = get_input_param<float>(input);
-          break;
-        case nvinfer1::DataType::kHALF:
-          param[i] = get_input_param<Eigen::half>(input);
-          break;
-        default:  // nvinfer1::DataType::kINT32:
-          param[i] = get_input_param<int>(input);
+      const auto& dims = input.GetTrtDims();
+      if (dims.nbDims != 1 || dims.d[0] != 1) {
+        return errors::InvalidArgument("Dimension for '", InputSpec()[i].name,
+                                       "' of ", node_def.op(), " operator ",
+                                       "should be equal to 1");
       }
     }
 
-    if ((delta_ = param[2]) == 0) {
-      return errors::InvalidArgument("The delta parameter of ", node_def.op(),
-                                     " operation cannot be equal to 0");
-    }
+    if (all_weights_) {
+      const auto num_intervals_float =
+          (param[1] - (start_ = param[0])) / delta_;
+      if (num_intervals_float < 0) {
+        const auto error = convert_range_error_msg(start_, param[1], delta_);
+        return errors::InvalidArgument(error);
+      }
 
-    const auto num_intervals_float = (param[1] - (start_ = param[0])) / delta_;
-    if (num_intervals_float < 0) {
-      const auto error = convert_range_error_msg(start_, param[1], delta_);
-      return errors::InvalidArgument(error);
+      num_values_ = static_cast<int>(num_intervals_float);
+      if (start_ + delta_ * num_values_ != param[1]) {
```

Comment:
When we have float input, exact equality is not guaranteed due to round off errors. Please update the condition, or the way the number of elements are calculated.

Reply:
In fact, we are not relying on exact equality here. Quite the contrary, we will NOT increment `num_values_` when exact equality occurs.